File: Pancake_Sort/Pancake_Sort.php
<?php
//function which actually sorts the array
function pancakeSort(&$arr, $num)    
{ 
    //start with whole array as input and reduces size by 1 on every iteration
    for ($currentSize = $num; $currentSize > 1; $currentSize--)  
    { 
        //finds the index of the maximum element in the array
        $maxIndex = findMaxIndex($arr, $currentSize);  
         //if the maximum index is not at the end of the current array then move the element to the end 
        if ($maxIndex != $currentSize-1)     
        { 
            //moves maximum number to the beginning
            reverseArray($arr, $maxIndex);
            //moves maximum number to the end by reversing the array    
            reverseArray($arr, $currentSize-1);  
        } 
    } 
} 

//reverses the array from index 0 to index 
function reverseArray(&$arr, $index)  
{ 
    $startIndex = 0; 
    //reverses till the startindex is not greater than index provided
    while ($startIndex < $index)   
    {  
        //swapping of elements
        $temp = $arr[$startIndex];        
        $arr[$startIndex] = $arr[$index]; 
        $arr[$index] = $temp; 
        //increasing starting Index of an array
        $startIndex++;      
        //decreasing ending index of an array
        $index--;           
    } 
} 

//Find the index of the maximum element in an array
function findMaxIndex($arr, $num)   
{ 
    $maxIndex = 0; 
    for ($i = 0; $i < $num; $i++) 
    {
        if ($arr[$i] > $arr[$maxIndex])
        {
            $maxIndex = $i;
        }
    }              
    return $maxIndex; 
}
  
$arr = array(112,65,12,56,100,93,20); 
//Counts the number of elements in the array
$num = count($arr);    
  
pancakeSort($arr, $num); 
  
echo("Sorted Array is:\n"); 
//Prints the sorted array
for($i = 0;$i < $num; $i++)    
{
    print($arr[$i]." "); 
}
  
return 0; 

/*
Input: 
  112,65,12,56,100,93,20

Output:
  12,20,56,65,93,100,112
*/
?>

File: Pancake_Sort/Pancake_Sort.kt
// Kotlin Code for Pancake Sort

class PancakeSorting
{   
    // This functions reverses the array upto index n
    fun reverseArray ( int:a [] , int:n ):void
    {
        int temp[] = new int [n+1] ;   
        for (i in 0 until n)
        {
            temp[i] = a[n-i];
        }
        for (i in 0 until n)
        {
        	a[i] = temp[i];
        }
    }
    
     // Function to find index of max element upto index n
     fun findMax (int:a[] , int:n ):int
     {
         int max = a[0] ; int j = 0;
         for (i in 1 until n)
         {  
             if (a[i] > max)
             {
                 max = a[i];
                 j = i;
             }
         }
         return j; 
      }
    
    fun pancakeSort (int:a [] , int:n):void
    {  
        int currentLength;
        for (currentLength in n until 0)
        {
            int j = findmax (a , currentLength-1);
            if (j != currentLength)
            {
                reverseArray (a , j);
                reverseArray (a , currentLength-1);
            }
        }
     }
    
    fun main ()
    {
        var read = Scanner(System.`in`)
        int i , j , k , l , m , n; 
        val n = read.nextLine().toInt()
        int array [] = new int [n];
        for(i in 0 until n)
        {
       	    array[i] = read.nextLine().toInt()
        }
        pancakesort (array, n);
        for (i in 0 until n)
        {
            println( array[i] + " " );	
        }
    }
}

/* 
Input
5  4  3  2  1 
Output 
1 2 3 4 5 
*/
